archiver/direct.py

```diff
@@ -77,6 +89,8 @@ def __archive_ena_entities(self, ingest_entities_to_update, submission):
         return ena_accessions
 
     def __exchange_sample_and_project_accessions(self, submission, biosample_accessions: list, biostudies_accession):
+        if isinstance(biostudies_accession, list):
```

Comment:
I would consider creating a utiity function `first_element_or_self`. This could be used in other locations as well.